extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddSidecarDecorator.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.kubernetes.deployment;
+
+import io.dekorate.deps.kubernetes.api.model.ObjectMeta;
+import io.dekorate.deps.kubernetes.api.model.PodSpecBuilder;
+import io.dekorate.kubernetes.config.Container;
+import io.dekorate.kubernetes.decorator.Decorator;
+import io.dekorate.kubernetes.decorator.NamedResourceDecorator;
+import io.dekorate.kubernetes.decorator.ResourceProvidingDecorator;
+
+/**
+ * Copied from dekorate in order to fix some issues
+ */
+class AddSidecarDecorator extends NamedResourceDecorator<PodSpecBuilder> {
```

Comment:
The class seems identical to the upstream version. Can you please point out the difference?

Reply:
It uses our custom `ContainerAdapter`, that's the only change